### DISCUSSION TITLE
Generate models from JSON schema?

### DISCUSSION THREAD

**Discussion by jods4** [christianhelle/refitter#312] [Ideas]
I'd like to generate C# models from JSON schema.
I was about to hard-code a small CLI tool to invoke `NJsonSchema` but then I thought that Refitter is already doing it, more or less:
- You already have CLI parsing, options loading.
- You generate code with `NSwag`, which is based on `NJsonSchema` so options are actually the same.

Would you entertain the idea of adding a command line switch to just generate a C# model from a JSON schema rater than a a full interface from OpenApi?

**Comment by christianhelle** [comment]
I'm not against it. We could introduce a CLI argument like --contracts-only or something like that.

**Reply by jods4** [reply]
Sounds like an interesting idea.

In my case I'm really working with just a JSON schema, not an OpenAPI spec. 
So for your code to not have to adjust to different input formats, you'll probably want a specific option?
```
refitter ./schema.json --json-schema

# Options that still make sense for this:
-s, --settings-file                           
-n, --namespace
-o, --output
--no-auto-generated-header
--internal
--no-banner
```

The codeGeneratorSettings would apply identically, as well as a few general properties in settings file:
```
{
  "openApiPath": "/path/to/your/openAPI", // Re-purposed for json schema path
  "namespace": "Org.System.Service.Api.GeneratedCode",
  "addAutoGeneratedHeader": true,
  "typeAccessibility": "Public",
  "outputFolder": "../CustomOutput",
  "outputFilename": "RefitInterface.cs",
  "codeGeneratorSettings": { 
     // These already are NJsonSchema options, they're applicable
     // See https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.CSharp/CSharpGeneratorSettings.cs
  }
}
```
